software/base/src/main/java/org/apache/brooklyn/entity/software/base/InboundPortsUtils.java

```diff
@@ -77,12 +82,17 @@
             Set<ConfigKey<?>> configKeys = Sets.newHashSet(extraConfigKeys);
             configKeys.addAll(entity.getEntityType().getConfigKeys());
             // Also add dynamically added config keys
-            configKeys.addAll(((EntityInternal)entity).config().getBag().getAllConfigAsConfigKeyMap().keySet());
+            Map<ConfigKey<?>, ?> configuredConfigKeys = ImmutableMap.copyOf(((EntityInternal) entity).config().getBag().getAllConfigAsConfigKeyMap());
+            configKeys.addAll(configuredConfigKeys.keySet());
```

Comment:
First, I started to use directly the `((EntityInternal) entity).config().getBag().getAllConfigAsConfigKeyMap()` but I found an exception when this tests was executed, because the entity was accessed by another thread (I found and `InterruptedException` what threw a NPE. 
Then, I decided to use some intermediate element, I used an immutable because the Map should not be modified. Does it make sense?